Book.cpp

```diff
@@ -311,3 +311,167 @@ void Book::getAllBooks() {
 Book::~Book()
 {
 }
+//Check if this works or needs any improvement or change
+//The code has been compiling successfully so far
+//Just see if the code for file search and delete is correct
+//Feel free to suggest any changes whatsover, always welcoming them.
+
+fstream f;
+struct booksIssued
+{
+	char LibraryID[100];
+	char ISBN[13];
+};
+ 
+class Employee
+{
+private:
+	char employeeID[10];
+	char membershipID[10];
+	booksIssued Books[100];
+	int bookCount;
+
+public:
+	Employee(){
+		memset(&employeeID,'\0',sizeof(employeeID));
+		memset(&membershipID,'\0',sizeof(membershipID));
+		bookCount = 0;
+		// addEmployee();
+
+	}
+	int getBookCount(){
+		return bookCount;
+	}
+	
+	char* getEmployeeID(){
+		return employeeID;
+	}
+
+	char* getMemberID(){
+		return membershipID;
+	}
+
+	booksIssued* getBooksIssued(){
+		return Books;
+	}
+
+	// void alterEmployeeID(){
+	// 	cin.get(employeeID,10);
+	// }
+
+	// void alterMembershipID(){
+	// 	cin.get(membershipID,10);
+	// }
+
+	void addEmployee(Employee& obj){
+		// char* temp = getMemberID();
+		// cin.get(obj.employeeID,10);
+		cin>>obj.employeeID;
+		// cin.get(obj.membershipID,10);
+		cin>>obj.membershipID;
+		if(existingEmployee() == false){		//no existing ID of the employee
+			f.open("EmployeeData.dat",ios::out|ios::app|ios::binary);
+			f.write((char*)&obj,sizeof(obj));
+			f.close();
+			cout<<"Employee Added"<<endl;
+		}
+		else{
+			cout<<"Already Registered."<<endl;
+		}
+	}
+
+	void deleteEmployee(Employee& obj){
+		fstream fin;
+		fstream fout;
+		Employee temp;
+		fin.open("temp.dat",ios::in|ios::out|ios::binary);
+		fout.open("EmployeeData.dat",ios::in|ios::out|ios::app|ios::binary);
+		fin.seekg(0,ios::beg);
+		while(fin.read((char*)&temp,sizeof(temp))){
+			if(strcmpi(obj.getMemberID(),temp.getMemberID()) == 0){
+				cout<<obj.employeeID<<" "<<temp.employeeID;
+				continue;
+			}
+			else{
+				fin.write((char*)&temp,sizeof(temp));
+			}
+		}
+		fin.close();
+		fout.close();
+		remove("EmployeeData.dat");
+		rename("temp.dat","EmployeeData.dat");
+
+	}
+
+	void addBook(Employee& b){
+		int nBooks;
+		cout<<"Number of books being issued: ";
+		cin>>nBooks;
+		while(nBooks--){
+			cout<<"1"<<endl;
+		++bookCount;
+		int idx = bookCount - 1;
+		cout<<"idx: "<<idx<<endl;
+		cin>>b.Books[idx].LibraryID;
+		cin>>b.Books[idx].ISBN;
+		}
+		fstream ftemp;
+		ftemp.open("temp.dat", ios::in | ios::out | ios::binary | ios::app);
+		f.open("EmployeeData.dat", ios::in | ios::binary);

```

Comment:
Did i miss something, i cant find the declaration for fstream f, i only see it for ftemp.